### DISCUSSION TITLE
Reolink Duo 2 Wifi IPC_529B17B8MP

### DISCUSSION THREAD

**Discussion by rx700** [AT0myks/reolink-fw-archive#1] [Firmware issues]
Currently there are 2 available Firmwares- to use the latest (v3.0.0.1391_22101061) you have to rename the file as follows:
IPC_529B17B8MP.1391_22101061.Reolink-Duo-2-WiFi.OV05A10.8MP.WIFI8812.REOLINK.pak

Minor changes to the version 1337- but beware: for me the Infrared-Light has a bug after midnight- it stays off sometimes. So perhaps you should stay on 1337.

**Comment by AT0myks** [comment]
Thank you for your feedback. I assume you gave the file the same name as in the screenshot from the source.

I have a vague memory of seeing a screenshot of the HTML of the browse button showing the accepted file extensions.
Could you share the same thing for your camera?
In your camera settings go to the page where you update the firmware, right click the browse button and click inspect, and then expand the `label` to show the `input`.

You should get something like this:
![image](https://user-images.githubusercontent.com/49416996/214145897-1b97de0f-274f-4ca3-b7fe-3e741b595888.png)

As you can see here the accepted file extensions are `.pak` and `.paks`. I wonder if your camera has something like `.REOLINK.pak`.
I could probably figure this out by looking at the HTML files inside the firmware, but maybe this comment can help someone.

**Reply by rx700** [reply]
Thanks for your reply- my camera has this as accepted extensions:
.pak,.paks,.paks,.pak

Perhaps it depends on the part in front of the dot, it looks like there is a filename-check after the upload of the file.

**Reply by AT0myks** [reply]
Alright thanks, I just realised this wouldn't make sense because you could simply edit the HTML to make it accept any file and if your camera had `.REOLINK.pak` this means (assuming you didn't edit the HTML) that you wouldn't even have had the opportunity to try the first file name because the firmware wouldn't have showed up in your file explorer.

As you say the check is done in the back.
I'm wondering if it has something to do with the model name and/or hardware version appearing in the file name. 
But I don't think the check is as good as I think it is (as in looking inside the firmware for an indication of a match, which would completely remove the need for a specific file name) because I'm pretty sure some users managed to install a firmware for an 8 camera NVR on a 16 camera NVR.
And this would also explain why simply renaming the file works.
